test/c3/appframework/appframework_test.go

```diff
@@ -308,4 +308,102 @@ var _ = Describe("c3appfw test", func() {
+			appFileList := testenv.GetAppFileList(appList, 1)
+
+			// Download ES App from S3
+			err := testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appFileList)
+			Expect(err).To(Succeed(), "Unable to download ES app file")
+
+			// Upload ES app to S3
+			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
+			Expect(err).To(Succeed(), "Unable to upload ES app to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Create App framework Spec
+			volumeName := "appframework-test-volume-" + testenv.RandomDNSName(3)
+			volumeSpec := []enterprisev1.VolumeSpec{testenv.GenerateIndexVolumeSpec(volumeName, testenv.GetS3Endpoint(), testenvInstance.GetIndexSecretName(), "aws", "s3")}
+
+			appSourceDefaultSpec := enterprisev1.AppSourceDefaultSpec{
+				VolName: volumeName,
+				Scope:   "cluster",
+			}
+			appSourceName := "appframework-" + testenv.RandomDNSName(3)
+			appSourceSpec := []enterprisev1.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceName, s3TestDir, appSourceDefaultSpec)}
+			appFrameworkSpec := enterprisev1.AppFrameworkSpec{
+				Defaults:             appSourceDefaultSpec,
+				AppsRepoPollInterval: 60,
+				VolList:              volumeSpec,
+				AppSources:           appSourceSpec,
+			}
+
+			// Create Single site Cluster and SHC, with App Framework enabled on SHC Deployer
+			// Deploy the CM
+			deployment.DeployClusterMaster(deployment.GetName(), "", "")
+
+			// Deploy the indexer cluster
+			indexerReplicas := 3
+			deployment.DeployIndexerCluster(deployment.GetName()+"-idxc", deployment.GetName(), indexerReplicas, deployment.GetName(), "")
+
+			// Deploy the SHC
+			shSpec := enterprisev1.SearchHeadClusterSpec{
+				CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+					Spec: splcommon.Spec{
+						ImagePullPolicy: "Always",
+					},
+					ExtraEnv: []corev1.EnvVar{
+						{
+							Name:  "SPLUNK_ES_SSL_ENABLEMENT",
+							Value: "ignore"},
+					},
+					Volumes: []corev1.Volume{},
+					ClusterMasterRef: corev1.ObjectReference{
+						Name: deployment.GetName(),
+					},
+					LivenessInitialDelaySeconds:  1450,
+					ReadinessInitialDelaySeconds: 1450,
+				},
+				Replicas:           3,
+				AppFrameworkConfig: appFrameworkSpec,
+			}
+			_, err = deployment.DeploySearchHeadClusterWithGivenSpec(deployment.GetName()+"-shc", shSpec)
+			Expect(err).To(Succeed(), "Unable to deploy SHC with App framework")
+
+			// Ensure that the CM goes to Ready phase
+			testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Ensure Indexers go to Ready phase
+			testenv.SingleSiteIndexersReady(deployment, testenvInstance)
+
+			// Ensure SHC go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify RF SF is met
+			testenv.VerifyRFSFMet(deployment, testenvInstance)
+
+			// Verify Apps are downloaded by init-container
+			esApp := "splunk-enterprise-security_640.spl"
```

Comment:
This line is redundant as these values are obtained in line 317. You can use the AppInfo variable to get this  value.

Reply:
To piggyback on Twain's comment this can be done by
`downloadedapps := append(testenv.GetAppFileList(appListV1, 1), appList[0])`

Reply:
`VerifyAppsDownloadedByInitContainer` requires the file with its extension, in this case "splunk-enterprise-security_640.spl". If passing "SplunkEnterpriseSecuritySuite", it'll fail

Reply:
In that case it's better to use the Map to get this value instead of declaring this independently. That way we preserve 1 source of truth.